### PR TITLE
Add celestia-da tutorial, network agnostic

### PR DESCRIPTION
**Pull request by yarikbratashchuk**
## Summary by CodeRabbit

- **Documentation**
	- Updated the tutorial on deploying a gm-world rollup on Celestia's data availability network, including setup and configuration instructions.
	- Revised the tutorial for building a sovereign rollup to use Rollkit CLI and a mocked DA network instead of Ignite CLI and Celestia.
	- Updated sidebar navigation in documentation to consolidate links under a new "Celestia" section.

### CODE REVIEW

**Review by MSevey — APPROVED** [review]
Awesome!

File: tutorials/celestia-da.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+Now, obtain an authentication token for your light node as follows (for mainnet beta, simply omit the --p2p.network flag):
+
+```bash
+AUTH_TOKEN=$(celestia light auth write --p2p.network mocha)
+echo -e "\n Your DA AUTH_TOKEN is $AUTH_TOKEN \n"
+```
+
+The output would look like this:
+
+```
```
Comment (jcstein) [inline tutorials/celestia-da.md:75]:
```suggestion
```bash
```

File: tutorials/celestia-da.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+## 🧹 Cleaning Previous Chain History
+
+From the [GM World Rollup](/tutorials/gm-world) tutorial, you should already have the `gmd` binary and the `$HOME/.gm` directory.
+
+To clear old rollup data:
+
+```bash
+rm -r /usr/local/bin/gmd && rm -rf $HOME/.gm
+
```
Comment (jcstein) [inline tutorials/celestia-da.md:29]:
```suggestion
```

File: tutorials/celestia-da.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
+# GM world rollup: Deploying to Celestia  
+
+## 🌞 Introduction {#introduction}
+
+This tutorial serves as a comprehensive guide for deploying your gm-world rollup on Celestia's data availability network. From the Rollkit perspective, there's no difference in posting blocks to Celestia's testnet or mainnet beta.
```
Comment (jcstein) [inline tutorials/celestia-da.md:5]:
```suggestion
This tutorial serves as a comprehensive guide for deploying your gm-world rollup on Celestia's data availability network. From the Rollkit perspective, there's no difference in posting blocks to Celestia's testnets or Mainnet Beta.
```

File: tutorials/celestia-da.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+## 🪶 Runnning a Celestia Light Node
+
+Before you can start your rollup node, you need to initiate, sync, and possibly fund a light node on one of Celestia's networks:
+
+- [Arabica Devnet](https://docs.celestia.org/nodes/arabica-devnet#arabica-devnet)
+- [Mocha Testnet](https://docs.celestia.org/nodes/mocha-testnet#mocha-testnet)
+- [Mainnet Beta](https://docs.celestia.org/nodes/mainnet#mainnet-beta)
+
+The main difference lies in how you fund your wallet address: using testnet tokens or [TIA](https://docs.celestia.org/learn/tia#overview-of-tia) for the mainnet beta.
```
Comment (jcstein) [inline tutorials/celestia-da.md:17]:
```suggestion
The main difference lies in how you fund your wallet address: using testnet tokens or [TIA](https://docs.celestia.org/learn/tia#overview-of-tia) for Mainnet Beta.
```

File: tutorials/celestia-da.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
+# GM world rollup: Deploying to Celestia  
+
+## 🌞 Introduction {#introduction}
+
+This tutorial serves as a comprehensive guide for deploying your gm-world rollup on Celestia's data availability network. From the Rollkit perspective, there's no difference in posting blocks to Celestia's testnet or mainnet beta.
+
+Before proceeding, ensure that you have completed the [GM World Rollup](/tutorials/gm-world) tutorial, which covers setting up a local sovereign gm-world rollup and connecting it to a local mock DA node.
+
+## 🪶 Runnning a Celestia Light Node
```
Comment (jcstein) [inline tutorials/celestia-da.md:9]:
```suggestion
## 🪶 Runnning a Celestia light node
```

File: tutorials/celestia-da.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+- [Arabica Devnet](https://docs.celestia.org/nodes/arabica-devnet#arabica-devnet)
+- [Mocha Testnet](https://docs.celestia.org/nodes/mocha-testnet#mocha-testnet)
+- [Mainnet Beta](https://docs.celestia.org/nodes/mainnet#mainnet-beta)
+
+The main difference lies in how you fund your wallet address: using testnet tokens or [TIA](https://docs.celestia.org/learn/tia#overview-of-tia) for the mainnet beta.
+
+After successfully starting a light node, it's time to start posting the blocks of data that your rollup generates.
+
+## 🧹 Cleaning Previous Chain History
```
Comment (jcstein) [inline tutorials/celestia-da.md:21]:
```suggestion
## 🧹 Cleaning previous chain history
```

File: tutorials/celestia-da.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+From the [GM World Rollup](/tutorials/gm-world) tutorial, you should already have the `gmd` binary and the `$HOME/.gm` directory.
```
Comment (jcstein) [inline tutorials/celestia-da.md:23]:
```suggestion
From the [GM world rollup](/tutorials/gm-world) tutorial, you should already have the `gmd` binary and the `$HOME/.gm` directory.
```

File: tutorials/celestia-da.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+To clear old rollup data:
+
+```bash
+rm -r /usr/local/bin/gmd && rm -rf $HOME/.gm
+
+```
+
+## 🏗️ Building Your Rollup
```
Comment (jcstein) [inline tutorials/celestia-da.md:32]:
```suggestion
## 🏗️ Building your rollup
```

File: tutorials/celestia-da.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+
+Now we need to rebuild our rollup by simply running the existing `init.sh` script:
+
+```bash
+cd $HOME/gm && bash init.sh
+```
+
+This process creates a new `$HOME/.gm` directory and a new `gmd` binary. Next, we need to connect our rollup to the running Celestia light node.
+
+## 🛠️ Configuring Flags for DA
```
Comment (jcstein) [inline tutorials/celestia-da.md:42]:
```suggestion
## 🛠️ Configuring flags for DA
```

File: tutorials/celestia-da.md [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+
+- `--rollkit.da_start_height`
+- `--rollkit.da_auth_token`
+- `--rollkit.da_namespace`
+
+Let's determine what to provide for each of them.
+
+First, let's query the DA Layer start height using an RPC endpoint provided by Celestia Labs. For Mocha testnet it would be - https://rpc-mocha.pops.one/block, and for mainnet beta - https://rpc.lunaroasis.net/block
+
+Here is an example for the Mocha testnet (replace URL for mainnet beta if needed):
```
Comment (jcstein) [inline tutorials/celestia-da.md:54]:
```suggestion
Here is an example for the Mocha testnet (replace URL for mainnet beta if needed):

```

File: tutorials/celestia-da.md [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+echo -e "\n Your DA_BLOCK_HEIGHT is $DA_BLOCK_HEIGHT \n"
+```
+
+You will see the output like this:
+
+```bash
+ Your DA_BLOCK_HEIGHT is 1777655
+```
+
+Now, obtain an authentication token for your light node as follows (for mainnet beta, simply omit the --p2p.network flag):
```
Comment (jcstein) [inline tutorials/celestia-da.md:66]:
```suggestion
Now, obtain an authentication token for your light node as follows (for Mainnet Beta, simply omit the --p2p.network flag):
```

File: tutorials/celestia-da.md [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+
+```bash
+AUTH_TOKEN=$(celestia light auth write --p2p.network mocha)
+echo -e "\n Your DA AUTH_TOKEN is $AUTH_TOKEN \n"
+```
+
+The output would look like this:
```
Comment (jcstein) [inline tutorials/celestia-da.md:73]:
```suggestion
The output will look like this:
```

File: tutorials/celestia-da.md [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+Now, let's also set up a namespace for our blocks by simply setting a variable like this:
+
+```bash
+DA_NAMESPACE=00000000000000000000000000000000000000000008e5f679bf7116cb
+```
+
+:::tip
+`00000000000000000000000000000000000000000008e5f679bf7116cb` is a default namespace for mocha testnet. You can set your own by using a command
```
Comment (jcstein) [inline tutorials/celestia-da.md:86]:
```suggestion
`00000000000000000000000000000000000000000008e5f679bf7116cb` is a default namespace for Mocha testnet. You can set your own by using a command
```

File: tutorials/celestia-da.md [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+similar to this (or, you could get creative 😎):
+
+```bash
+openssl rand -hex 10
+```
+
+Replace the last 10 characters in `00000000000000000000000000000000000000000008e5f679bf7116cb` with the newly generated 10 characters.
```
Comment (jcstein) [inline tutorials/celestia-da.md:93]:
```suggestion
Replace the last 20 characters (10 bytes) in `00000000000000000000000000000000000000000008e5f679bf7116cb` with the newly generated 10 characters.
```

File: tutorials/celestia-da.md [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+
+[Learn more about namespaces](https://docs.celestia.org/developers/node-tutorial#namespaces).
+:::
+
+## 🔥 Running Your Rollup Connected to a Celestia Light Node
```
Comment (jcstein) [inline tutorials/celestia-da.md:98]:
```suggestion
## 🔥 Running your rollup connected to a Celestia light node
```

File: tutorials/celestia-da.md [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+
+Before you can start your rollup node, you need to initiate, sync, and possibly fund a light node on one of Celestia's networks:
+
+- [Arabica Devnet](https://docs.celestia.org/nodes/arabica-devnet#arabica-devnet)
+- [Mocha Testnet](https://docs.celestia.org/nodes/mocha-testnet#mocha-testnet)
+- [Mainnet Beta](https://docs.celestia.org/nodes/mainnet#mainnet-beta)
+
+The main difference lies in how you fund your wallet address: using testnet tokens or [TIA](https://docs.celestia.org/learn/tia#overview-of-tia) for the mainnet beta.
+
+After successfully starting a light node, it's time to start posting the blocks of data that your rollup generates.
```
Comment (jcstein) [inline tutorials/celestia-da.md:19]:
```suggestion
After successfully starting a light node, it's time to start posting the batches of blocks of data that your rollup generates.
```

File: tutorials/celestia-da.md [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+```bash
+gmd start \
+    --rollkit.aggregator \
+    --rollkit.da_auth_token $AUTH_TOKEN \
+    --rollkit.da_namespace $DA_NAMESPACE \
+    --rollkit.da_start_height $DA_BLOCK_HEIGHT \
+    --minimum-gas-prices="0.025stake"
+```
+
+Now, the rollup is running and posting blocks (aggregated in batches) to Celestia. You can view your rollup by finding your namespace or account on [Mocha devnet](https://docs.celestia.org/nodes/mocha-testnet#explorers) or [mainnet beta](https://docs.celestia.org/nodes/mainnet#explorers) explorers.
```
Comment (jcstein) [inline tutorials/celestia-da.md:111]:
```suggestion
Now, the rollup is running and posting blocks (aggregated in batches) to Celestia. You can view your rollup by finding your namespace or account on [Mocha testnet](https://docs.celestia.org/nodes/mocha-testnet#explorers) or [mainnet beta](https://docs.celestia.org/nodes/mainnet#explorers) explorers.
```

File: tutorials/celestia-da.md [hunk 18]
```diff
@@ -0,0 +1,119 @@
...
+
+::: info
```
Comment (jcstein) [inline tutorials/celestia-da.md:113]:
```suggestion
::: info
```

File: tutorials/gm-world.md [hunk 19]
```diff
@@ -68,7 +68,7 @@ If you get errors of `gmd` not found, you may need to add the `go/bin` directory
 :::
 
 ```bash
-curl -sSL https://rollkit.dev/install-gm-rollup.sh | sh
+cd $HOME && curl -sSL https://rollkit.dev/install-gm-rollup.sh | sh
```
Comment (jcstein) [inline tutorials/gm-world.md:71]: do we need to do this @yarikbratashchuk ? I think it looks much cleaner as:
```
curl -sSL https://rollkit.dev/install-gm-rollup.sh | sh
```

but would be ideal if it is not actually downloaded using something similar to:

```bash
bash -c "$(curl -sL https://docs.celestia.org/celestia-app.sh)"
```

File: tutorials/gm-world.md [hunk 20]
```diff
@@ -68,7 +68,7 @@ If you get errors of `gmd` not found, you may need to add the `go/bin` directory
...
 ```
 
 ## 🚀 Starting Your Rollup {#start-your-rollup}
```
Comment (jcstein) [inline tutorials/gm-world.md:74]:
```suggestion
## 🚀 Starting your rollup {#start-your-rollup}
```

**Review by jcstein — CHANGES_REQUESTED** [review]
small changes

File: tutorials/celestia-da.md (hunk 18 above)
Comment (jcstein) [inline tutorials/celestia-da.md:113]:
```suggestion
::: info
```

File: tutorials/celestia-da.md [hunk 21]
```diff
@@ -0,0 +1,119 @@
+# GM world rollup: Deploying to Celestia  
+
+## 🌞 Introduction {#introduction}
+
+This tutorial serves as a comprehensive guide for deploying your gm-world rollup on Celestia's data availability network. From the Rollkit perspective, there's no difference in posting blocks to Celestia's testnets or Mainnet Beta.
```
Comment (jcstein) [inline tutorials/celestia-da.md:5]:
```suggestion
This tutorial serves as a comprehensive guide for deploying your GM world rollup on Celestia's data availability network. From the Rollkit perspective, there's no difference in posting blocks to Celestia's testnets or Mainnet Beta.
```

File: tutorials/celestia-da.md [hunk 22]
```diff
@@ -0,0 +1,119 @@
+# GM world rollup: Deploying to Celestia  
+
+## 🌞 Introduction {#introduction}
+
+This tutorial serves as a comprehensive guide for deploying your GM world rollup on Celestia's data availability network. From the Rollkit perspective, there's no difference in posting blocks to Celestia's testnets or Mainnet Beta.
+
+Before proceeding, ensure that you have completed the [GM World Rollup](/tutorials/gm-world) tutorial, which covers setting up a local sovereign gm-world rollup and connecting it to a local mock DA node.
```
Comment (jcstein) [inline tutorials/celestia-da.md:7]:
```suggestion
Before proceeding, ensure that you have completed the [GM world rollup](/tutorials/gm-world) tutorial, which covers setting up a local sovereign gm-world rollup and connecting it to a local mock DA node.
```

File: tutorials/celestia-da.md [hunk 23]
```diff
@@ -0,0 +1,119 @@
...
+
+This tutorial serves as a comprehensive guide for deploying your GM world rollup on Celestia's data availability network. From the Rollkit perspective, there's no difference in posting blocks to Celestia's testnets or Mainnet Beta.
+
+Before proceeding, ensure that you have completed the [GM world rollup](/tutorials/gm-world) tutorial, which covers setting up a local sovereign gm-world rollup and connecting it to a local mock DA node.
+
+## 🪶 Running a Celestia light node
+
+Before you can start your rollup node, you need to initiate, sync, and possibly fund a light node on one of Celestia's networks:
+
+- [Arabica Devnet](https://docs.celestia.org/nodes/arabica-devnet#arabica-devnet)
```
Comment (jcstein) [inline tutorials/celestia-da.md:13]:
```suggestion
- [Arabica Devnet](https://docs.celestia.org/nodes/arabica-devnet)
```

File: tutorials/celestia-da.md [hunk 24]
```diff
@@ -0,0 +1,119 @@
...
+- [Mocha Testnet](https://docs.celestia.org/nodes/mocha-testnet#mocha-testnet)
```
Comment (jcstein) [inline tutorials/celestia-da.md:14]:
```suggestion
- [Mocha Testnet](https://docs.celestia.org/nodes/mocha-testnet)
```

File: tutorials/celestia-da.md [hunk 25]
```diff
@@ -0,0 +1,119 @@
...
+- [Mainnet Beta](https://docs.celestia.org/nodes/mainnet#mainnet-beta)
```
Comment (jcstein) [inline tutorials/celestia-da.md:15]:
```suggestion
- [Mainnet Beta](https://docs.celestia.org/nodes/mainnet)
```

File: tutorials/celestia-da.md [hunk 26]
```diff
@@ -0,0 +1,119 @@
+# GM world rollup: Deploying to Celestia  
+
+## 🌞 Introduction {#introduction}
+
+This tutorial serves as a comprehensive guide for deploying your GM world rollup on Celestia's data availability network. From the Rollkit perspective, there's no difference in posting blocks to Celestia's testnets or Mainnet Beta.
```
Comment (jcstein) [inline tutorials/celestia-da.md:5]:
```suggestion
This tutorial serves as a comprehensive guide for deploying your GM world rollup on Celestia's data availability (DA) network. From the Rollkit perspective, there's no difference in posting blocks to Celestia's testnets or Mainnet Beta.
```

File: tutorials/celestia-da.md [hunk 27]
```diff
@@ -0,0 +1,119 @@
...
+
+Now, the rollup is running and posting blocks (aggregated in batches) to Celestia. You can view your rollup by finding your namespace or account on [Mocha testnet](https://docs.celestia.org/nodes/mocha-testnet#explorers) or [mainnet beta](https://docs.celestia.org/nodes/mainnet#explorers) explorers.
+
+::: info
+For details on configuring gas prices specifically for the DA network, see our [DA Network Gas Price Guide](/guides/gas-price). This is separate from the `--minimum-gas-prices="0.025stake"` setting, which is used for rollup network operations.
+:::
+
+## 🎉 Next steps
+
+Congratulations! You've built a local rollup that posts to Celestia's testnet or mainnet. Well done! Now, go forth and build something great! Good luck!
```
Comment (jcstein) [inline tutorials/celestia-da.md:119]:
```suggestion
Congratulations! You've built a local rollup that posts to Celestia's testnets or mainnet. Well done! Now, go forth and build something great! Good luck!
```

File: tutorials/gm-world.md [hunk 28]
```diff
@@ -1,5 +1,5 @@
 ---
-description: Build a sovereign rollup with Ignite CLI, Celestia, and Rollkit locally and on a testnet
+description: Build a sovereign rollup using only Rollkit CLI and mocked DA network
```
Comment (jcstein) [inline tutorials/gm-world.md:2]:
```suggestion
description: Build a sovereign rollup using only Rollkit CLI and a mock DA network.
```

File: tutorials/gm-world.md [hunk 29]
```diff
@@ -1,5 +1,5 @@
...
 ---
 
 # GM World Rollup
```
Comment (jcstein) [inline tutorials/gm-world.md:5]:
```suggestion
# GM world rollup
```

File: tutorials/gm-world.md [hunk 30]
```diff
@@ -68,10 +68,10 @@ If you get errors of `gmd` not found, you may need to add the `go/bin` directory
 :::
 
 ```bash
-curl -sSL https://rollkit.dev/install-gm-rollup.sh | sh
+cd $HOME && curl -sSL https://rollkit.dev/install-gm-rollup.sh | sh
```
Comment (jcstein) [inline tutorials/gm-world.md:71]:
```suggestion
cd $HOME && bash -c "$(curl -sSL https://rollkit.dev/install-gm-rollup.sh)"
```

File: tutorials/gm-world.md [hunk 31]
```diff
@@ -68,10 +68,11 @@ If you get errors of `gmd` not found, you may need to add the `go/bin` directory
 :::
 
 ```bash
-curl -sSL https://rollkit.dev/install-gm-rollup.sh | sh
+cd $HOME
```
Comment (jcstein) [inline tutorials/gm-world.md:71]: @yarikbratashchuk I think this needs sudo access. [see without sudo logs](https://app.warp.dev/block/oD5RrlXSSpcteFcEpVXuXU)

File: tutorials/gm-world.md [hunk 32]
```diff
@@ -68,10 +68,11 @@ If you get errors of `gmd` not found, you may need to add the `go/bin` directory
...
+bash -c "$(curl -sSL https://rollkit.dev/install-gm-rollup.sh)"
```
Comment (jcstein) [inline tutorials/gm-world.md:72]:
```suggestion
sudo bash -c "$(curl -sSL https://rollkit.dev/install-gm-rollup.sh)"
```

File: tutorials/gm-world.md (hunk 31 above)
Comment (yarikbratashchuk) [reply tutorials/gm-world.md:71]: changed init.sh script to go install, so we don't need sudo now, as it's installed into the path
sudo is another dependancy we would require to install

Comment (jcstein) [reply tutorials/gm-world.md:71]: sounds good 🫡

Comment (jcstein) [inline tutorials/gm-world.md:71]:
```suggestion
cd $HOME && bash -c "$(curl -sSL https://rollkit.dev/install-gm-rollup.sh)"
```

File: tutorials/gm-world.md (hunk 32 above)
Comment (jcstein) [inline tutorials/gm-world.md:72]:
```suggestion
```

**Review by jcstein — APPROVED** [review]
:shipit:

**Pull request merged**